.github/workflows/ci.yml

```diff
@@ -0,0 +1,38 @@
+name: test
+on:
+  pull_request:
+    branches: [ main ]
+jobs:
+  codegen:
+    name: Codegen
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Restore go build cache #This action allows caching dependencies and build outputs to improve workflow execution time 
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.mod') }}
+      - name: Setup Go Environment #https://github.com/marketplace/actions/setup-go-environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.17.1' # The Go version to download (if necessary) and use
```

Comment:
This is great. I think we may forget to update this version whenever we update the [.go-version](https://github.com/hashicorp/go-tfe/blob/main/.go-version). 

what do you think about adding a extra step in here that outputs the content of [.go-version](https://github.com/hashicorp/go-tfe/blob/main/.go-version) and ensures it is `1.17.1`. That way, if someone changes .go-version, this step will fail and we will know to update the Go version here as well.

Reply:
NIce! I love the suggestion